zerver/views/messages.py

```diff
@@ -320,6 +323,19 @@ def highlight_string_text_offsets(text, locs):
     result = u''
     for loc in locs:
         (offset, length) = loc
+
+        if offset < pos:
+            continue
+
+        # When the offset is inside a tag, we want to move it back, so that it
+        # highlights the whole tag, not just some attribute of it
+        last_tag_open = string.rfind('<', pos, offset+length)
```

Comment:
Can be made a function. There seems like code duplication with the code block starting at line 357. With a separate function, you can easily write unit tests.